File: javascript/js/account.js
function Account(owner, accNum, balance) {
    // 객체 속성
    this.owner = owner; // account.owner = "홍길동";
    this.accNum = accNum;
    this.balance = balance;
    this.name = "직장인 비과세 예금";
    this.bankname = "우리은행";

    //객체 기능(메서드)
    this.deposit = function (money) {
        // 예금
        this.balance += money;
    };
    
    this.checkBalance = function(){
        // 잔액조회
        return this.balance;
    };

    this.withdraw = function (money) {
        //인출
        if (this.balance >= money){
            this.balance -= money;
        } else {
            window.alert("잔액이 부족합니다.");
        }
    };
}

File: java/JV13_Constructor/src/ex1/Account.java
package ex1;

public class Account {
	private int balance = 0;
	
	public void deposit(int money) {
		if (isValid(money)) {
			balance += money;			
		} else {
			System.out.println("유효하지 않은 요청입니다.");
		}
	}
	
	public void withdraw(int money) {
		if (isValid(money)) {
			if (isValid(balance - money)) {
				balance -= money;
			} else {
				System.out.println("유효하지 않은 요청입니다.");
			}
		} else {
			System.out.println("유효하지 않은 요청입니다.");
		}
	}
	
	public int getBalance() {
		return balance;
	}
	
	private boolean isValid(int money) {
//		boolean result = false;
//		if (money >= 0) {
//			result = true;
//		}
//		return result;
		return money >= 0;
	}
}
